### PR TITLE
parser: implement Restore for DeleteTableList

### PR DESCRIPTION
**Pull request by wangbo**
pingcap/tidb#8532

### CODE REVIEW

File: ast/dml_test.go [hunk 1]
```diff
@@ -139,3 +139,13 @@ func (tc *testDMLSuite) TestLimitRestore(c *C) {
...
 
+func (tc *testDMLSuite) TestDeleteTableListRestore(c *C) {
+	testCases := []NodeRestoreTestCase{
+		{"t1,t2", "`t1`,`t2`"},
+	}
+	extractNodeFunc := func(node Node) Node {
+		return node.(*DeleteStmt).Tables
+	}
+	RunNodeRestoreTest(c, testCases, "DELETE %s FROM t1 INNER JOIN t2 INNER JOIN t3 WHERE t1.id=t2.id AND t2.id=t3.id;", extractNodeFunc)
+	RunNodeRestoreTest(c, testCases, "DELETE FROM %s USING t1 INNER JOIN t2 INNER JOIN t3 WHERE t1.id=t2.id AND t2.id=t3.id;", extractNodeFunc)
```
Comment (kennytm) [inline ast/dml_test.go]: The INNER JOIN stuff doesn't appear necessary. 

`DELETE %s FROM t1, t2` and `DELETE FROM %s USING t1, t2` should be sufficient.

File: ast/dml.go [hunk 2]
```diff
@@ -234,7 +234,15 @@ type DeleteTableList struct {
...
 // Restore implements Node interface.
 func (n *DeleteTableList) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+    for i, t := range n.Tables {
+		if err := t.Restore(ctx); err != nil {
+			return errors.Annotate(err, "An error occurred while restore DeleteTableList.Tablename")
+		}
+		if i != len(n.Tables) - 1 {
+			ctx.WritePlain(",")
+		}
```
Comment (kennytm) [inline ast/dml.go]: You could place the `,` check above `Restore`.

```go
if i != 0 {
    ctx.WritePlain(",")
}
if err := t.Restore(ctx); err != nil {
    ...
}
```

File: ast/dml.go [hunk 3]
```diff
@@ -234,7 +234,15 @@ type DeleteTableList struct {
 
 // Restore implements Node interface.
 func (n *DeleteTableList) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+    for i, t := range n.Tables {
+		if i != 0 {
+			ctx.WritePlain(",")
+		}
+    	if err := t.Restore(ctx); err != nil {
+			return errors.Annotate(err, "An error occurred while restore DeleteTableList.Tablename")
```
Comment (leoppro) [inline ast/dml.go]: could you record index in error log?

**Comment by leoppro** [comment]
please exec `gofmt -s -l -w ast/dml.go`

**Pull request merged**